Yesod/Auth/OAuth2/Github.hs

```diff
@@ -47,12 +49,14 @@ instance FromJSON GithubUser where
     parseJSON _ = mzero
 
 data GithubUserEmail = GithubUserEmail
-    { githubUserEmail :: Text
+    { githubUserEmailEmail :: Text
```

Comment:
> `GithubUser` would have a field `emails :: [GithubUserEmail]`

I think this would be cool, but it's a little tricky because the basic profile info that's currently in `GithubUser` comes from a different web request as the emails. I'd be fine with making this change, but I think it would be best to do in another pull request.

I think the `GithubUserEmailEmail` to `GithubUserEmailAddress` changes would be good to make. If you'd like to make those, I'd be happy to merge this in.